[2022-11-14 03:56:11,388] {taskinstance.py:826} INFO - Dependencies all met for <TaskInstance: pyspark_bmi.download_dataset 2022-11-13T12:00:00+00:00 [queued]>
[2022-11-14 03:56:11,407] {taskinstance.py:826} INFO - Dependencies all met for <TaskInstance: pyspark_bmi.download_dataset 2022-11-13T12:00:00+00:00 [queued]>
[2022-11-14 03:56:11,407] {taskinstance.py:1017} INFO - 
--------------------------------------------------------------------------------
[2022-11-14 03:56:11,407] {taskinstance.py:1018} INFO - Starting attempt 1 of 1
[2022-11-14 03:56:11,407] {taskinstance.py:1019} INFO - 
--------------------------------------------------------------------------------
[2022-11-14 03:56:11,417] {taskinstance.py:1038} INFO - Executing <Task(PythonOperator): download_dataset> on 2022-11-13T12:00:00+00:00
[2022-11-14 03:56:11,421] {standard_task_runner.py:51} INFO - Started process 36680 to run task
[2022-11-14 03:56:11,425] {standard_task_runner.py:75} INFO - Running: ['airflow', 'tasks', 'run', 'pyspark_bmi', 'download_dataset', '2022-11-13T12:00:00+00:00', '--job-id', '19', '--pool', 'default_pool', '--raw', '--subdir', 'DAGS_FOLDER/pyspark_bmi.py', '--cfg-path', '/tmp/tmpe3k4zeul']
[2022-11-14 03:56:11,427] {standard_task_runner.py:76} INFO - Job 19: Subtask download_dataset
[2022-11-14 03:56:11,474] {logging_mixin.py:103} INFO - Running <TaskInstance: pyspark_bmi.download_dataset 2022-11-13T12:00:00+00:00 [running]> on host 676a9425fa5f
[2022-11-14 03:56:11,520] {taskinstance.py:1232} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=airflow
AIRFLOW_CTX_DAG_ID=pyspark_bmi
AIRFLOW_CTX_TASK_ID=download_dataset
AIRFLOW_CTX_EXECUTION_DATE=2022-11-13T12:00:00+00:00
AIRFLOW_CTX_DAG_RUN_ID=backfill__2022-11-13T12:00:00+00:00
[2022-11-14 03:56:12,292] {taskinstance.py:1396} ERROR - write() argument must be str, not bytes
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1086, in _run_raw_task
    self._prepare_and_execute_task_with_callbacks(context, task)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1260, in _prepare_and_execute_task_with_callbacks
    result = self._execute_task(context, task_copy)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1300, in _execute_task
    result = task_copy.execute(context=context)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/operators/python.py", line 117, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/operators/python.py", line 128, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/pyspark_bmi.py", line 25, in download_dataset
    open("/opt/airflow/datasets/weight_height_index.csv", "w+").write(response.content)
TypeError: write() argument must be str, not bytes
[2022-11-14 03:56:12,294] {taskinstance.py:1440} INFO - Marking task as FAILED. dag_id=pyspark_bmi, task_id=download_dataset, execution_date=20221113T120000, start_date=20221114T035611, end_date=20221114T035612
[2022-11-14 03:56:12,318] {local_task_job.py:118} INFO - Task exited with return code 1
